implementations/rust/ockam/ockam_transport_tcp/tests/send_receive.rs

```diff
@@ -55,33 +50,27 @@ impl Worker for Echoer {
 #[allow(non_snake_case)]
 #[ockam_macros::test]
 async fn tcp_lifecycle__reconnect__should_not_error(ctx: &mut Context) -> Result<()> {
-    let rand_port = rand::thread_rng().gen_range(10000, 65535);
-    let bind_address = format!("127.0.0.1:{}", rand_port);
-    let bind_address = bind_address.as_str();
-
     ctx.start_worker("echoer", Echoer).await?;
 
     let transport = TcpTransport::create(ctx).await?;
-    transport.listen(bind_address).await?;
+    let listener_address = transport.listen("127.0.0.1:0").await?.to_string();
 
     let mut child_ctx = ctx.new_context(Address::random_local()).await?;
-    let msg: String = {
-        let mut rng = rand::thread_rng();
-        iter::repeat(())
-            .map(|()| rng.sample(&rand::distributions::Alphanumeric))
-            .take(10)
-            .collect()
-    };
+    let msg: String = rand::thread_rng()
+        .sample_iter(&rand::distributions::Alphanumeric)
+        .take(256)
+        .map(char::from)
+        .collect();
 
-    let tx_address = transport.connect(bind_address).await?;
+    let tx_address = transport.connect(&listener_address).await?;
 
-    let r = route![(TCP, format!("localhost:{}", rand_port)), "echoer"];
+    let r = route![(TCP, listener_address.clone()), "echoer"];
```

Comment:
`SocketAddr` implements `Copy`, so in theory the clone call is not needed.

Reply:
It expects a `String` there, and also a few lines above that (when calling `transport.connect(&listener_address)`). It looks like cloning that string is the less painful option.

Reply:
Ah sorry! You are right of course, I missed that you had it converted to a string again in line 56.